spec/lib/active_record_utils_spec.rb

```diff
@@ -27,20 +26,10 @@
       expect(tx_result).to be false
     end
 
-    it 'adds DB_UPDATE_FAILED result and returns false when the transaction raises ActiveRecordError' do
-      expect(audit_results).to receive(:add_result).with(
-        AuditResults::DB_UPDATE_FAILED, a_string_matching('ActiveRecord::InvalidForeignKey')
-      )
-      tx_result = described_class.with_transaction_and_rescue(audit_results) do
-        PreservationPolicy.default_policy.delete
-      end
-      expect(tx_result).to be false
-    end
```

Comment:
@jmartin-sul Not sure if worth replacing? 🤷🏻‍♂️

Reply:
iirc, this behavior is worth testing, so i'd be inclined to replace (i'm not sure if we have other parts of the code that test this in a more integration-y way for real transactions in the app).

i think we could probably update this by doing something like creating a `CompleteMoab` using factorybot, and trying to delete its `MoabStorageRoot`, wrapped in `with_transaction_and_rescue(audit_results)`.